src/wallet/wallet2.cpp

```diff
@@ -4236,6 +4410,373 @@ static uint32_t get_count_above(const std::vector<wallet2::transfer_details> &tr
+      }
+    }
+    
+    if(!add_transfer)
+      continue;
+    
+    MTRACE("Adding output " << o.public_key);
+    m_transfers.push_back(boost::value_initialized<transfer_details>());
+    transfer_details& td = m_transfers.back();
+    
+    td.m_block_height = o.height;
+    td.m_global_output_index = o.global_index;
+    td.m_txid = txid;
+    
+
+    
+    // Add to extra
+    add_tx_pub_key_to_extra(td.m_tx, tx_pub_key);
+    
+    td.m_key_image = unspent_key_image;
+    td.m_key_image_known = !m_watch_only;
+    td.m_amount = boost::lexical_cast<uint64_t>(o.amount);
+    td.m_pk_index = 0; //TODO: Not sure what this is?
+    td.m_internal_output_index = o.index; //TODO:: is this correct?
+    td.m_spent = spent;
+
+    tx_out txout;
+    txout.target = txout_to_key(public_key);
+    txout.amount = td.m_amount;
+    
+    // Add empty outs to match m_internal_output_index
+    // TODO: better solution? why is vout a vector and not a map?
+    //for(size_t i = 0; i < td.m_internal_output_index ; ++i)
+    //td.m_tx.vout.push_back(txout_empty);
+    td.m_tx.vout.resize(td.m_internal_output_index + 1);
+    td.m_tx.vout[td.m_internal_output_index] = txout;
+    
+    
+   if (!o.rct.empty()) //TODO: are all outputs with a rct value acutally a rct output?
+   {
+      // o.rct is a string with length 64+64+64 (<rct commit> + <encrypted mask> + <rct amount>)
+      // We decrypt the mask and save it to td.m_mask.
+      rct::key mm_rct_commit;
+      rct::key encrypted_mask;
+      string_tools::hex_to_pod(o.rct.substr(0,64), mm_rct_commit);
+      string_tools::hex_to_pod(o.rct.substr(64,64), encrypted_mask);
+      
+      crypto::key_derivation derivation;
+      generate_key_derivation(tx_pub_key, get_account().get_keys().m_view_secret_key, derivation);
+      crypto::secret_key scalar;
+      crypto::derivation_to_scalar(derivation, td.m_internal_output_index, scalar);
+      rct::key decrypted_mask;
+      sc_sub(decrypted_mask.bytes,encrypted_mask.bytes,rct::hash_to_scalar(rct::sk2rct(scalar)).bytes);
+
+      rct::key new_rct_commit = rct::commit(td.amount(),decrypted_mask);
+      
+      // MDEBUG("Dumping rct mask stuff");
+      // MDEBUG(string_tools::pod_to_hex(encrypted_mask));
+      // MDEBUG(string_tools::pod_to_hex(decrypted_mask));
+      // MDEBUG(td.m_internal_output_index);
+      // MDEBUG(string_tools::pod_to_hex(derivation));
+      // MDEBUG(string_tools::pod_to_hex(scalar));
+      // MDEBUG(string_tools::pod_to_hex(rct::hash_to_scalar(rct::sk2rct(scalar))));
+      
+      // Make sure rct commit is the same 
+      if(!(mm_rct_commit == new_rct_commit)){ 
+        MERROR("decrypted mask is wrong. rct commit: " << string_tools::pod_to_hex(mm_rct_commit) << " != " << string_tools::pod_to_hex(new_rct_commit));
+        //TODO: Throw
+        assert(false);
+      }
+      
+      td.m_mask = decrypted_mask;
+      td.m_rct = true;
+    }
+    // TODO: howto check if tx is a miner_tx with mymonero api? 
```

Comment:
`coinbase` is only available in `get_address_txs` it seems. I'm currently using `get_unspent_outs` to generate `m_transfers` in `wallet2` because i need the key images. Would it be possible to add `coinbase` to `get_unspent_outs`?